hepdata/modules/records/templates/hepdata_records/components/resource_details.html

```diff
@@ -34,7 +34,14 @@ <h4>
       {% elif ctx.display_type == 'link' %}
         <p><a href="{{ ctx.resource.file_location }}">{{ ctx.resource.file_location }}</a></p>
       {% elif ctx.display_type == 'code' %}
-        <textarea id="code-contents" height="70%">{{ ctx.contents }}</textarea>
+
+        {% if ctx.contents and ctx.contents != 'Large text file' %}
+          <textarea id="code-contents" height="70%">{{ ctx.contents }}</textarea>
+        {% else %}
+        <div id="code-contents-fail" height="70%">
+          <p style="font-weight:bolder">This file ({{ ctx.filesize }}MB) is larger than our loading threshold (1.0MB), and is only available for download below.</p>
```

Comment:
Better to avoid hard-coding the threshold ("1.0MB") and add a space before "MB" (twice):
```
This file ({{ ctx.filesize }} MB) is larger than our loading threshold ({{ ctx.ADDITIONAL_SIZE_LOAD_CHECK_THRESHOLD }} MB)
```